src/metriculous/evaluators/_classification_figures_bokeh.py

```diff
@@ -141,11 +141,15 @@ def _bokeh_confusion_matrix(
     """
 
     def figure() -> Figure:
-        cm = sklmetrics.confusion_matrix(y_true, y_pred)
+        cm = sklmetrics.confusion_matrix(y_true, y_pred, labels=list(range(len(class_names))))
         cm_normalized = cm.astype("float") / cm.sum()
         cm_normalized_by_pred = cm.astype("float") / cm.sum(axis=0, keepdims=True)
         cm_normalized_by_true = cm.astype("float") / cm.sum(axis=1, keepdims=True)
 
+        cm_normalized = np.nan_to_num(cm_normalized)
+        cm_normalized_by_pred = np.nan_to_num(cm_normalized_by_pred)
+        cm_normalized_by_true = np.nan_to_num(cm_normalized_by_true)
```

Comment:
I'm a bit surprised that Bokeh doesn't handle the NaNs properly... I thought I had a similar issue a while ago and it was resolved by making sure all fields in the `ColumnDataSource` are numpy arrays, as opposed to lists of floats. A few lines further down all the lists get converted to arrays, so I'm a bit puzzled.

Is your virtualenv up to date?

Unless we have to, I'd like to avoid converting NaNs to make sure none of the information displayed in the tooltips is misleading.

Reply:
![image](https://user-images.githubusercontent.com/9771046/102633678-a5cc1c00-4148-11eb-8fff-58aeaf88f7c3.png)

Reply:
Hmm interesting, I am using bokeh 2.2.3 which should be the latest and I am getting this:

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-13-934bfb0cc871> in <module>
      2 print(normalized.shape)
      3 print(gt_1hot.shape)
----> 4 mo.compare_classifiers(
      5     ground_truth=gt_1hot.argmax(-1),
      6     model_predictions=[normalized],

~/Library/Caches/pypoetry/virtualenvs/document-classification-ReJIVt9a-py3.8/lib/python3.8/site-packages/metriculous/_comparison.py in display(self, include_spacer, width)
     50 
     51         for row in _figure_rows(self.evaluations, include_spacer=include_spacer):
---> 52             plotting.show(row)
     53 
     54         # noinspection PyBroadException

~/Library/Caches/pypoetry/virtualenvs/document-classification-ReJIVt9a-py3.8/lib/python3.8/site-packages/bokeh/io/showing.py in show(obj, browser, new, notebook_handle, notebook_url, **kw)
    137         return run_notebook_hook(state.notebook_type, 'app', obj, state, notebook_url, **kw)
    138 
--> 139     return _show_with_state(obj, state, browser, new, notebook_handle=notebook_handle)
    140 
    141 #-----------------------------------------------------------------------------

~/Library/Caches/pypoetry/virtualenvs/document-classification-ReJIVt9a-py3.8/lib/python3.8/site-packages/bokeh/io/showing.py in _show_with_state(obj, state, browser, new, notebook_handle)
    171 
    172     if state.notebook:
--> 173         comms_handle = run_notebook_hook(state.notebook_type, 'doc', obj, state, notebook_handle)
    174         shown = True
    175 

~/Library/Caches/pypoetry/virtualenvs/document-classification-ReJIVt9a-py3.8/lib/python3.8/site-packages/bokeh/io/notebook.py in run_notebook_hook(notebook_type, action, *args, **kw)
    296     if _HOOKS[notebook_type][action] is None:
    297         raise RuntimeError("notebook hook for %r did not install %r action" % notebook_type, action)
--> 298     return _HOOKS[notebook_type][action](*args, **kw)
    299 
    300 #-----------------------------------------------------------------------------

~/Library/Caches/pypoetry/virtualenvs/document-classification-ReJIVt9a-py3.8/lib/python3.8/site-packages/bokeh/io/notebook.py in show_doc(obj, state, notebook_handle)
    506     from ..embed.notebook import notebook_content
    507     comms_target = make_id() if notebook_handle else None
--> 508     (script, div, cell_doc) = notebook_content(obj, comms_target)
    509 
    510     publish_display_data({HTML_MIME_TYPE: div})

~/Library/Caches/pypoetry/virtualenvs/document-classification-ReJIVt9a-py3.8/lib/python3.8/site-packages/bokeh/embed/notebook.py in notebook_content(model, notebook_comms_target, theme)
     83     # has models with the same IDs as they were started with
     84     with OutputDocumentFor([model], apply_theme=theme, always_new=True) as new_doc:
---> 85         (docs_json, [render_item]) = standalone_docs_json_and_render_items([model])
     86 
     87     div = div_for_render_item(render_item)

~/Library/Caches/pypoetry/virtualenvs/document-classification-ReJIVt9a-py3.8/lib/python3.8/site-packages/bokeh/embed/util.py in standalone_docs_json_and_render_items(models, suppress_callback_warning)
    292     docs_json = {}
    293     for doc, (docid, _) in docs.items():
--> 294         docs_json[docid] = doc.to_json()
    295 
    296     render_items = []

~/Library/Caches/pypoetry/virtualenvs/document-classification-ReJIVt9a-py3.8/lib/python3.8/site-packages/bokeh/document/document.py in to_json(self)
    899         # this is a total hack to go via a string, needed because
    900         # our BokehJSONEncoder goes straight to a string.
--> 901         doc_json = self.to_json_string()
    902         return loads(doc_json)
    903 

~/Library/Caches/pypoetry/virtualenvs/document-classification-ReJIVt9a-py3.8/lib/python3.8/site-packages/bokeh/document/document.py in to_json_string(self, indent)
    928         }
    929 
--> 930         return serialize_json(json, indent=indent)
    931 
    932     def validate(self):

~/Library/Caches/pypoetry/virtualenvs/document-classification-ReJIVt9a-py3.8/lib/python3.8/site-packages/bokeh/core/json_encoder.py in serialize_json(obj, pretty, indent, **kwargs)
    164         indent = 2
    165 
--> 166     return json.dumps(obj, cls=BokehJSONEncoder, allow_nan=False, indent=indent, separators=separators, sort_keys=True, **kwargs)
    167 
    168 

~/.pyenv/versions/3.8.5/lib/python3.8/json/__init__.py in dumps(obj, skipkeys, ensure_ascii, check_circular, allow_nan, cls, indent, separators, default, sort_keys, **kw)
    232     if cls is None:
    233         cls = JSONEncoder
--> 234     return cls(
    235         skipkeys=skipkeys, ensure_ascii=ensure_ascii,
    236         check_circular=check_circular, allow_nan=allow_nan, indent=indent,

~/.pyenv/versions/3.8.5/lib/python3.8/json/encoder.py in encode(self, o)
    197         # exceptions aren't as detailed.  The list call should be roughly
    198         # equivalent to the PySequence_Fast that ''.join() would do.
--> 199         chunks = self.iterencode(o, _one_shot=True)
    200         if not isinstance(chunks, (list, tuple)):
    201             chunks = list(chunks)

~/.pyenv/versions/3.8.5/lib/python3.8/json/encoder.py in iterencode(self, o, _one_shot)
    255                 self.key_separator, self.item_separator, self.sort_keys,
    256                 self.skipkeys, _one_shot)
--> 257         return _iterencode(o, 0)
    258 
    259 def _make_iterencode(markers, _default, _encoder, _indent, _floatstr,

ValueError: Out of range float values are not JSON compliant
```

Reply:
Strange... I was expecting this to be fixed with 38e16198f3fc98701084b676f1f42a5b13adceaf in August. Any chance you branched out from an old version that did not contain that fix? It would be interesting to see if the problem persists after rebasing onto the latest `develop`. Also worth noting that the latest release (0.2.1) does not yet contain that fix.

Reply:
Rebasing should also fix the CI. I fixed the CI problems on `develop` after you opened this PR.

Reply:
The fix is also included in version 0.3.0 which I just released. Let me know if the problem persists, I'd still love to find out what's going on there.